### ISSUE TITLE
[Bug] Multiple-Word-Search: result presentation is depending on upper-/lower-case of first word.

### ISSUE THREAD

**Issue by evoha** [dotnet/docfx#8322]
**Describe the bug**
When searching with multiple words, there is a strange behavior. Depending on how the _first_ (!) word is written regarding upper-/lower-case, the presentation (excerpt) of the result is different.
Note: The first search result is always the same article, but the excerpt does not show the part where the search term is used. Therefore the user might not consider this as a valid result and won't click on it.

**To Reproduce**

Here are some examples:

My documentation contains an article which contains the text 'WordA WordB'.

Search for 'WordA – WordB' --> OK

Search for 'WordA – wordB' --> OK
Search for 'wordA – WordB' --> **NOK**: first result is the same, but the excerpt shows the wrong part.

Search for 'WordA – wORDB' --> OK
Search for 'wORDA – WordB' --> **NOK**: first result is the same, but the excerpt shows the wrong part.

Search for 'WordC WordD … WordE' --> OK
Search for 'WordC wORDD … WordE' --> OK
Search for 'wordC WordD … WordE' --> **NOK**: first result is the same, but the excerpt shows the wrong part.
Search for 'WordC wORDD … wORDE' --> OK

**To sum up:** If the first word differs in the upper-/lower-case (in comparison to the text in the article), the excerpt of the result is presented different. Upper-/Lower-case of second and further words do not matter.
Special characters seem to be essential for reproducing the issue. I was not able to reproduce this issue with searches without ' – ' or ' ... '.

**Expected behavior**
The search should not be case sensitive in general.

**Context (please complete the following information):**
- Browser: edge
- Form factor: desktop
- Docfx version: docfx.console.2.59.4
- Template:default
